src/main.rs

```diff
@@ -133,50 +133,56 @@ fn create_short_id(req: &mut Request) -> Result<Response> {
 }
 
+fn handle_get(req: &Request) -> Result<Response> {
+    // handle GET
+    if req.get_path() == "/" {
+        Ok(Response::from_status(StatusCode::OK).with_header(
+            header::SET_COOKIE,
+            format!(
+                "passcode={:?}; Secure; HttpOnly",
+                CONFIG_DIC
+                    .get("passcode")
+                    .ok_or_else(|| anyhow!("No passcode in config store"))?
+            ),
+        ))
+    } else {
+        match get_redirect_url(req) {
+            Ok(resp) => Ok(resp),
+            Err(e) => {
+                Ok(Response::from_status(StatusCode::NOT_FOUND)
+                    .with_body_text_plain(&e.to_string()))
+            }
+        }
+    }
+}
+
+fn handle_post(req: &mut Request) -> Response {
+    // handle POST
+    match create_short_id(req) {
+        Ok(resp) => resp,
+        Err(e) => {
+            Response::from_status(StatusCode::NOT_ACCEPTABLE).with_body_text_plain(&e.to_string())
+        }
+    }
+}
+
+fn handle_options() -> Response {
+    // handle OPTIONS
+    Response::from_status(StatusCode::NO_CONTENT)
+        .with_header(header::ALLOW, "GET, POST, OPTIONS")
+        .with_header(header::ACCESS_CONTROL_ALLOW_ORIGIN, "*")
+        .with_header(header::ACCESS_CONTROL_ALLOW_HEADERS, "*")
+        .with_header(header::ACCESS_CONTROL_ALLOW_METHODS, "*")
+}
+
 #[fastly::main]
 fn main(mut req: Request) -> Result<Response> {
-    let resp = match *req.get_method() {
-        // Get URL of a short ID
-        Method::GET => match req.get_path() {
-            // Root dir, send editor.html page to client with passcode in cookie
-            "/" => Response::from_status(StatusCode::OK)
-                .with_header(
-                    header::SET_COOKIE,
-                    format!(
-                        "passcode={}; Secure; HttpOnly",
-                        CONFIG_DIC
-                            .get("passcode")
-                            .ok_or_else(|| anyhow!("No passcode in config store"))?
-                    ),
-                )
-                .with_body_text_html(include_str!("editor.html")),
-
-            // redirect url request
-            _ => match get_redirect_url(&req) {
-                Ok(resp) => resp,
-                Err(e) => Response::from_status(StatusCode::NOT_FOUND)
-                    .with_body_text_plain(&e.to_string()),
-            },
-        },
-
-        // Create short ID for a URL
-        Method::POST => match create_short_id(&mut req) {
-            Ok(resp) => resp,
-            Err(e) => Response::from_status(StatusCode::NOT_ACCEPTABLE)
-                .with_body_text_plain(&e.to_string()),
-        },
-
-        // For CORS request, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS
-        Method::OPTIONS => Response::from_status(StatusCode::NO_CONTENT)
+    match *req.get_method() {
+        Method::GET => handle_get(&req),
+        Method::POST => Ok(handle_post(&mut req)),
```

Comment:
Can we make handle_post and handle_option also return Result<Response>,  like handle_get()? 
Looking forward to some consistency here.